src/components/HelFormFields/Selectors/TypeSelector.js

```diff
@@ -12,90 +12,61 @@ const {
-    }
+        const {type, isCreateView} = this.state;
+        const {editor} = this.props;
 
+        if (prevProps.editor.values.type_id !== editor.values.type_id && editor.values.type_id !== type) {
+            const newType = editor.values.type_id || EVENT_TYPE.GENERAL;
+            this.setState({type: newType});
 
-    handleMount = () => {
-        const {event:{type_id},editor:{values}} = this.props
-        const editedEventTypeIsEvent = type_id === EVENT_TYPE.GENERAL
-        const editedEventTypeIsCourse = type_id === EVENT_TYPE.COURSE
-        const editedEventTypeIsHobby = type_id === EVENT_TYPE.HOBBIES
-        let type = ''
-        if (editedEventTypeIsEvent) {
-            type = 'event'
-        }
-        if (editedEventTypeIsCourse) {
-            type = 'courses'
-        }
-        if (editedEventTypeIsHobby) {
-            type = 'hobby'
+        }else if (prevProps.editor.values && !editor.values.type_id && !isCreateView) {
+            this.setState({isCreateView: true});
+            this.context.dispatch(setData({type_id: EVENT_TYPE.GENERAL}))
         }
-        this.setState({type: type})
     }
 
-    /**
-     * Handles the updating of the state based on changes.
-     * @param prevState Previous state
-     */
-    handleUpdate = (prevState = {}) => {
-        const {type, isCreateView} = this.state
-        const {router} = this.context.store.getState()
-        const {editor:{values}, event} = this.props
-
-        // object containing the updated states
-        let stateToSet = {}
-
-        // whether we are creating a new event. used to help determine the radio disabled state
-        const updatedIsCreateView = get(router, ['location' ,'pathname'], '').includes('/event/create/new')
-        const creatingNewSubEvent = get(router, ['location', 'pathname'], '').includes('/recurring/add/')
-        const superEventIsNotNull = get(event, 'super_event_type') !== null
-        const editedEventTypeIsEvent = get(event, 'type_id') === EVENT_TYPE.GENERAL
-        const editedEventTypeIsCourse = get(event, 'type_id') === EVENT_TYPE.COURSE
-        const editedEventTypeIsHobby = get(event, 'type_id') === EVENT_TYPE.HOBBIES
-
-        // update the isCreateView according to whether we're creating a new event or updating an existing one
-        if (updatedIsCreateView !== isCreateView) {
-            stateToSet.isCreateView = updatedIsCreateView
-        }
 
-        if(updatedIsCreateView && superEventIsNotNull && !values.type_id && type === '') {
-            stateToSet.type = 'event'
-            this.context.dispatch(setData({type_id: EVENT_TYPE.GENERAL}))
+    handleMount = () => {
+        const {event:{type_id},editor:{values}} = this.props;
+        const {router} = this.context.store.getState();
+        // a brand event with no data sets this to true -> dispatches the initial type.
+        let dispatchInitialType = false;
+
+        const newState = {};
+        // editing existing event?
```

Comment:
Is this really a question? Really should be fixed that the purpose of the statement is readable at code level